pycbc/inject/inject.py

```diff
@@ -403,15 +404,22 @@ def make_strain_from_inj_object(self, inj, delta_t, detector_name,
             f_l = inj.f_lower
         else:
             f_l = f_lower
-
-        name, phase_order = legacy_approximant_name(inj.waveform)
-
-        # compute the waveform time series
-        hp, hc = get_td_waveform(
-            inj, approximant=name, delta_t=delta_t,
-            phase_order=phase_order,
-            f_lower=f_l, distance=inj.distance,
-            **self.extra_args)
+	approx_name=inj.waveform #put the path to the ASCII file as approximant
+	if approx_name.endwith(('.txt', '.dat')):
+		hp, hc, delta_t_ascii = get_waveform_from_ascii(inj.waveform)
+		if delta_t != delta_t_ascii:
+			raise ValueError("Sample rate of the ascii and the ini are different")
+		else:
+			hp = (hp/inj.distance)*100000 #reference distance for ASCII waveform is 100Mpc
```

Comment:
Any reason we can't set the distance as normal? I.E. normally distance is taken to be in Kpc for the xml format, and internally we use Mpc for everything. Why is this simply not * 100 ? Any reason the ASCII format needs to have this distance convention?